### PR TITLE
Fix: change multiline-always to proportional-always

### PR DESCRIPTION
**Pull request by Nokel81**
- If the node is multiline then require newline
- If the node is singleline then require space

Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Comment by ljharb** [comment]
Can you elaborate?

This is a breaking change with respect to #3260, but since it's not yet released, we could safely make it.

File: lib/rules/jsx-tag-spacing.js [hunk 1]
```diff
@@ -178,54 +180,68 @@ function validateAfterOpening(context, node, option) {
...
-            return fixer.insertTextBefore(closingToken, '\n');
-          },
-        });
-      }
-    }
-
-    if (leftToken.loc.start.line !== closingToken.loc.start.line) {
-      return;
-    }
+  if (node.selfClosing) {
```
Comment (ljharb) [inline lib/rules/jsx-tag-spacing.js:183]: let's revert this change to keep the diff minimal

File: lib/rules/jsx-tag-spacing.js [hunk 2]
```diff
@@ -178,54 +180,68 @@ function validateAfterOpening(context, node, option) {
...
+    return;
+  }
 
-    const adjacent = !sourceCode.isSpaceBetweenTokens(leftToken, closingToken);
+  const sourceCode = context.getSourceCode();
+  const leftToken = option === 'proportional-always'
+    ? getTokenBeforeClosingBracket(node)
+    : sourceCode.getLastTokens(node, 2)[0];
```
Comment (ljharb) [inline lib/rules/jsx-tag-spacing.js:190]: does `getLastTokens` always return an array?

Comment (Nokel81) [reply lib/rules/jsx-tag-spacing.js:190]: Yes it does, at least that is what the eslint docs imply.

**Review by ljharb — APPROVED** [review]

**Comment by burtek** [comment]
Docs not changed? https://github.com/jsx-eslint/eslint-plugin-react/issues/3294

**Comment by ljharb** [comment]
yes, they were overlooked.
